### PR TITLE
fix(argo-cd): Add usages to ArgoCD server certificate

### PR DESCRIPTION
**Pull request by ShimiTaNaka**
Issue ref: https://github.com/argoproj/argo-helm/issues/2142

Checklist:

* [V] I have bumped the chart version according to [versioning](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#versioning)
* [V] I have updated the documentation according to [documentation](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#documentation)
* [ ] I have updated the chart changelog with all the changes that come with this pull request according to [changelog](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#changelog).
* [V] Any new values are backwards compatible and/or have sensible default.
* [V] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md).
* [ ] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/stable/developer-guide/ci/)).

### CODE REVIEW

**Comment by yu-croco** [comment]
Hi @ShimiTaNaka , thank you for your PR.
Can you please fix failed CI and handle below?

> I have bumped the chart version according to [versioning](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#versioning)
> I have updated the documentation according to [documentation](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#documentation)

**Comment by ShimiTaNaka** [comment]
@yu-croco Thanks, I've bumped the version
however I'm not 100% sure why the PR title doesn't fit the linting

**Comment by yu-croco** [comment]
@ShimiTaNaka 
Thank you!

> however I'm not 100% sure why the PR title doesn't fit the linting

This is because PR accepts only specific title. Ref: https://github.com/argoproj/argo-helm/blob/main/.github/workflows/pr-title.yml#L27-L35
@tico24 resolved it, thank you!

**Comment by tico24** [comment]
...and it only took me two attempts!

**Comment by ShimiTaNaka** [comment]
@yu-croco I think it's good now, just not sure how to sort the DCO, I have added the signing to 2 of the commits however it still shows as if I haven't

**Comment by ShimiTaNaka** [comment]
@yu-croco I think it's ready for review (?)

**Comment by yu-croco** [comment]
@ShimiTaNaka 
Please update  (remove old one and add new one) [here](https://github.com/argoproj/argo-helm/pull/2143/files#diff-16f38cd1a4674cb682ac9f015fbc1c1ff552f024a8f791c16de0de21a1f65771L28-L30)

> I have updated the documentation according to [documentation](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#documentation)

File: charts/argo-cd/templates/argocd-server/certificate.yaml [hunk 1]
```diff
@@ -30,4 +30,10 @@ spec:
   privateKey:
     {{- toYaml . | nindent 4 }}
   {{- end }}
+  {{- if .Values.server.certificate.usages }}
+  usages:
+    {{- range .Values.server.certificate.usages }}
+    - {{ . | quote }}
+    {{- end }}
+  {{- end }}
```
Comment (yu-croco) [inline charts/argo-cd/templates/argocd-server/certificate.yaml:38]: Maybe this is simpler? 🙋 

```yaml
  {{- with .Values.server.certificate.usages }}
  usages:
    {{- toYaml . | nindent 4 }}
  {{- end }}
```

Comment (ShimiTaNaka) [reply charts/argo-cd/templates/argocd-server/certificate.yaml:38]: Probably, I wasn't sure actually, my first time writing helm.
I did see example in the code for the `range`

Comment (yu-croco) [reply charts/argo-cd/templates/argocd-server/certificate.yaml]: FYI
[here](https://github.com/argoproj/argo-helm/pull/2143/files#diff-6395eca2b798a2c7a98337b4f336f77d333f193264e733ff78a412faf08f4807R29-R32) is the existing sample.

Comment (ShimiTaNaka) [reply charts/argo-cd/templates/argocd-server/certificate.yaml]: Thanks :)
Updated, hope it's fine

**Review by yu-croco — APPROVED** [review]
Thank you for your contribution! LGTM.

**Review by tico24 — APPROVED** [review]

**Comment by ShimiTaNaka** [comment]
Thank you guys for the fast response!

**Pull request merged**